### ISSUE TITLE
Helper functions require argument when it should be given by final property

### ISSUE THREAD

**Issue by ndslusarz** [vehicle-lang/vehicle#396]
There is a problem in the python code - not a bug per se, it was apparently coded that way - where helper function in a property, such as `boundedByEpsilon` or any of the others in the spec below, immediately requires some argument provided when trying to train with the loss function giving this error (as far as I was able to tell). Python comments claim this should not be a problem and said functions are "wrapped in a lambda with no arguments to delay evaluation" so I'm not sure where the problem is exactly - this specification, either way, should not require any outside arguments other then datasets and quantifier sampling which are provided.

```
  File "e:/vehicle/vehicle/vehicle-python/tests/test_mnist.py", line 56, in train
    vehicle_loss = vehicle_batch_loss()
TypeError: <lambda>() missing 1 required positional argument: 'v'
```

The problem is the argument that this function should take is given to it in the finall loss property (here `robust1`).

The property in question is the edited mnist. The code to call python on it can be found on the `loss_experiments` branch in`vehicle-python/tests/test_mnist.py`
```
--------------------------------------------------------------------------------
-- Network

-- Declare the network used to classify images. The output of the network is a
-- score for each of the digits 0 to 9.
@network
mnist : Tensor Rat [28, 28] -> Vector Rat 10

--epsilon : Rat
--epsilon = 0.01

--delta : Rat
--delta = 0.02

-- Next we define what it means for an image `x` to be in a ball of
-- size epsilon around 0.
boundedByEpsilon : Tensor Rat [28, 28] -> Bool
boundedByEpsilon x = forall i j . -0.01 <= x ! i ! j <= 0.01

--The disjunction avoids having to define absolute value
boundedByDelta : Tensor Rat [28, 28] -> Tensor Rat [28, 28] -> Bool
boundedByDelta x y = forall i . (-0.02 <= (((mnist x) ! i ) - ((mnist y) ! i )) <= 0.02) 


robustSingleInput : Tensor Rat [28, 28] -> Bool
robustSingleInput image = forall pertubation .
    let perturbedImage = image - pertubation in
    boundedByEpsilon pertubation => boundedByDelta image perturbedImage

@property
robust1 : Bool
robust1 = forall x . robustSingleInput x

```

**Comment by MatthewDaggitt** [comment]
Will have a look at this as soon as the latest round of PRs merge in.

**Comment by MatthewDaggitt** [comment]
Fixed by #401
